### PR TITLE
[posix-netif] refactor address changing logs

### PR DESCRIPTION
**Pull request by simonlingoogle**

### CODE REVIEW

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -851,28 +851,31 @@ static void processTransmit(otInstance *aInstance)
...
-static void logAddrEvent(bool isAdd, bool isUnicast, struct sockaddr_in6 &addr6, otError error)
+static void logAddrEvent(bool isAdd, const ot::Ip6::Address &aAddress, otError error)
 {
     char addressString[INET6_ADDRSTRLEN + 1];
 
     // these parameters may not be used if logging is disabled at compile time
-    OT_UNUSED_VARIABLE(isUnicast);
-    OT_UNUSED_VARIABLE(addr6);
+    OT_UNUSED_VARIABLE(aAddress);
     OT_UNUSED_VARIABLE(addressString);
```
Comment (librasungirl) [inline src/posix/platform/netif.cpp:860]: nit-style: move forward to the beginning of this body

File: src/posix/platform/netif.cpp [hunk 2]
```diff
@@ -851,28 +851,31 @@ static void processTransmit(otInstance *aInstance)
...
+                          ? " (already subscribed, ignored)"
+                          : error == OT_ERROR_REJECTED ? " (rejected)"
+                                                       : error == OT_ERROR_NOT_FOUND ? " (not found, ignored)" : "");
     }
     else
     {
-        otLogWarnPlat("[netif] %s [%s] %s failed (%s)", isAdd ? "ADD" : "DEL", isUnicast ? "U" : "M",
-                      inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
-                      otThreadErrorToString(error));
+        otIp6AddressToString(&aAddress, addressString, sizeof(addressString));
```
Comment (librasungirl) [inline src/posix/platform/netif.cpp:875]: nit: dup with above Line 865, may move out of `if-else`

File: src/posix/platform/netif.cpp [hunk 3]
```diff
@@ -851,28 +851,31 @@ static void processTransmit(otInstance *aInstance)
...
 
-    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY)) ||
+    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY || error == OT_ERROR_REJECTED)) ||
         ((!isAdd) && (error == OT_ERROR_NOT_FOUND)))
     {
-        otLogNotePlat("[netif] %s [%s] %s%s", isAdd ? "ADD" : "DEL", isUnicast ? "U" : "M",
-                      inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
-                      error == OT_ERROR_ALREADY ? " (already subscribed, ignored)"
-                                                : error == OT_ERROR_NOT_FOUND ? " (not found, ignored)" : "");
+        otIp6AddressToString(&aAddress, addressString, sizeof(addressString));
```
Comment (librasungirl) [inline src/posix/platform/netif.cpp:865]: may directly use `ToString()` method of class `ot::Ip6::Address`, or should the formal parameter be of type`otIp6Address`

**Review by librasungirl — APPROVED** [review]
👍

File: src/posix/platform/netif.cpp [hunk 4]
```diff
@@ -851,28 +851,24 @@ static void processTransmit(otInstance *aInstance)
...
-
-    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY)) ||
+    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY || error == OT_ERROR_REJECTED)) ||
         ((!isAdd) && (error == OT_ERROR_NOT_FOUND)))
     {
-        otLogNotePlat("[netif] %s [%s] %s%s", isAdd ? "ADD" : "DEL", isUnicast ? "U" : "M",
-                      inet_ntop(AF_INET6, addr6.sin6_addr.s6_addr, addressString, sizeof(addressString)),
-                      error == OT_ERROR_ALREADY ? " (already subscribed, ignored)"
-                                                : error == OT_ERROR_NOT_FOUND ? " (not found, ignored)" : "");
+        otLogInfoPlat("[netif] %s [%s] %s%s", isAdd ? "ADD" : "DEL", aAddress.IsMulticast() ? "M" : "U",
```
Comment (bukepo) [inline src/posix/platform/netif.cpp:861]:
```suggestion
        otLogNotePlat("[netif] %s [%s] %s%s", isAdd ? "ADD" : "DEL", aAddress.IsMulticast() ? "M" : "U",
```

**Review by bukepo — APPROVED** [review]

File: src/posix/platform/netif.cpp (hunk 4 above)
Comment (simonlingoogle) [reply src/posix/platform/netif.cpp:861]: Note means something unusual but fine. 

But I think the case should happen quite often. So, maybe INFO is better?

Comment (bukepo) [reply src/posix/platform/netif.cpp:861]: SGTM

File: src/core/net/netif.cpp [hunk 5]
```diff
@@ -399,7 +399,7 @@ Error Netif::SubscribeExternalMulticast(const Address &aAddress)
 
     for (const MulticastAddress *cur = &linkLocalAllRoutersAddress; cur; cur = cur->GetNext())
     {
-        VerifyOrExit(cur->GetAddress() != aAddress, error = kErrorInvalidArgs);
+        VerifyOrExit(cur->GetAddress() != aAddress, error = kErrorRejected);
```
Comment (abtink) [inline src/core/net/netif.cpp:402]: Should we also update the new `Rejected` error in `netif.hpp` docs for `SubscribeExternalMulticast()`?
Also in public OT header `ip6.h` for `otIp6SubscribeMulticastAddress()`?

**Review by abtink — APPROVED** [review]
Looks good. Thanks. 👍 
One small suggestions below to update the docs about the functions/methods now returning  `Rejected` error code.

File: src/core/net/netif.cpp (hunk 5 above)
Comment (simonlingoogle) [reply src/core/net/netif.cpp:402]: doc updated.

**Review by abtink — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**